Grab Current Tracing Fake Mode in a couple spots

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #99377

Fix for https://github.com/pytorch/pytorch/issues/99286. There were a couple locations we were instantiating new fake modes instead of grabbing the correct one from the current tracing context/inputs.

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

eellison committed:
Fix hf bart inference failure

[ghstack-poisoned]

eellison committed:
Update on "Fix hf bart inference failure"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

eellison committed:
Update on "Fix hf bart inference failure"




cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

eellison committed:
Update on "Grab Current Tracing Fake Mode in a couple spots"


Fix for https://github.com/pytorch/pytorch/issues/99286. There were a couple locations we were instantiating new fake modes instead of grabbing the correct one from the current tracing context/inputs.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

eellison committed:
Update on "Grab Current Tracing Fake Mode in a couple spots"


Fix for https://github.com/pytorch/pytorch/issues/99286. There were a couple locations we were instantiating new fake modes instead of grabbing the correct one from the current tracing context/inputs.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

ezyang commented on torch/_inductor/fx_passes/joint_graph.py:
```diff
@@ -14,7 +15,7 @@
     from .fuse_attention import _sfdp_init
 
-    with FakeTensorMode():
+    with detect_fake_mode() or FakeTensorMode():
```
This is wrong as the patterns here are shared globally so you shouldn't arbitrarily pick the fake mode from a particular run; I actually have the more correct fix on a branch I am working on

eellison replied:
Okay - I'm going to close and re-assign to you then.

vkuzo replied:
@ezyang  could you share which PR? I want to follow when it lands as I have this PR checked out locally to unblock some quantization experiments.

eellison commented:
@pytorchbot rebase

eellison committed:
Update on "Grab Current Tracing Fake Mode in a couple spots"


Fix for https://github.com/pytorch/pytorch/issues/99286. There were a couple locations we were instantiating new fake modes instead of grabbing the correct one from the current tracing context/inputs.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]
